screwdriver/screwdriver.go

```diff
@@ -137,34 +144,102 @@ func handleResponse(res *http.Response) ([]byte, error) {
 	return body, nil
 }
 
+func retry(attempts int, callback func() error) (err error) {
+	b := &backoff.Backoff{
+		Min:    2 * time.Second,
+		Max:    32 * time.Second,
+		Factor: 2,
+	}
+
+	for i := 0; ; i++ {
+		err = callback()
+		if err == nil {
+			return nil
+		}
+
+		if i >= (attempts - 1) {
+			b.Reset()
+			break
+		}
+
+		sleep(b.Duration())
+
+		log.Println("Retrying request to Screwdriver.")
+	}
+	return fmt.Errorf("after %d attempts, last error: %s", attempts, err)
+}
+
 func (a api) get(url *url.URL) ([]byte, error) {
 	req, err := http.NewRequest("GET", url.String(), nil)
 	if err != nil {
 		return nil, fmt.Errorf("generating request to Screwdriver: %v", err)
 	}
 	req.Header.Set("Authorization", tokenHeader(a.token))
 
-	res, err := a.client.Do(req)
+	res := &http.Response{}
+	resError := errors.New("")
+	err = retry(5, func() error {
+		res, resError = a.client.Do(req)
+		if resError != nil {
+			return nil
+		}
+		if res.StatusCode/100 == 5 {
+			return fmt.Errorf("response contains status code %v, retrying GET request", res.StatusCode)
```

Comment:
This isn't entirely accurate, is it? After 5 errors, iIt would result in something like this:

"timeout on request: response contains status code 500, retying GET request"

Would be nicer to have 2 things:

1) After failure, log.Printf("WARNING: received response 500 from GET http://api.screwdriver.cd/builds/123 retrying (1/5)")
2) When all requests are exhausted, return error:

"GET retries exhausted: 500 returned from GET http://api.screwdriver.cd/builds/123"